var/spack/repos/builtin/packages/parsec/package.py

```diff
@@ -104,27 +104,26 @@ def check(self):
                 warn += "https://bitbucket.org/icldistcomp/parsec/issues"
                 tty.msg(warn)
 
-    def test_parsec(self):
-        """Compile and run a user program with the installed library"""
-        with working_dir(join_path(self.install_test_root, "contrib/build_with_parsec")):
-
-            with test_part(
-                self, "test_parsec_cmake", purpose="Check if CMake can find PaRSEC and its targets"
-            ):
-                cmake = which("cmake")
-                cmake(".")
-
-            with test_part(self, "test_parsec_make", purpose="Check if tests can compile"):
-                make = which("make")
-                make()
-
-            programs = ["dtd_test_allreduce", "write_check"]
-            for program in programs:
-                with test_part(self, "test_parsec_" + program, purpose="Run " + program):
-                    exe = which("./" + program)
-                    exe()
+    def run_parsec(self, exe_name):
+        """Common parsec testing method"""
+        with working_dir(join_path(install_test_root(self), "contrib/build_with_parsec")):
```

Comment:
Stand-alone tests aren't supposed to be running under the installation prefix.

Try moving the `cmake`-`make` steps (within the `working_dir` context manager) to after `cache_extra_test_sources()` in `cache_test_sources()`.

Then see if it each of the tests can simply instantiate the `Executable` of the `exe_name` that is located in `join_path(install_test_root(self), "contrib", "build_with_parsec", exe_name)` (which could be done in their respective `test_*` methods).

Reply:
Nice try.  Unfortunately, this doesn't work in this case.

Based on test results, it looks like moving `cmake` and `make` to post-install doesn't work because the generated `CMakeCache.txt` file has hard-coded installation paths:

```
...
CMake Error: The current CMakeCache.txt directory $spack_test/edxkh7u6g3umd32mmnkzsytgd6e4cb6h/parsec-3.0.2209-ssokfzy/cache/parsec/contrib/build_with_parsec/CMakeCache.txt is different than the directory $spack/opt/spack/linux-rhel8-x86_64_v3/gcc-12.1.1/parsec-3.0.2209-ssokfzy64ygpbdvo4oc5dbsdcwr2blvh/.spack/test/contrib/build_with_parsec where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
FAILED: Parsec::test_contrib: Command exited with status 1:
...
```

@AcriusWinter Please use the more common approach in the `MyLibPackage` example in the docs https://spack.readthedocs.io/en/latest/packaging_guide.html#saving-build-and-install-time-files.